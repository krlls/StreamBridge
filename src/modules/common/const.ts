export const limits = {
  music: {
    pagination: {
      maxLimit: 50,
      maxOffset: 60,
    },
  },
}
