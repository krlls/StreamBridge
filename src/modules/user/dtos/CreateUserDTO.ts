export class CreateUserDTO {
  login: string
  pass: string

  constructor(login: string, pass: string) {
    this.login = login
    this.pass = pass
  }
}
