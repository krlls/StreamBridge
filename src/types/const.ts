const TYPES = {
  UserController: Symbol.for('UserController'),

  UserService: Symbol.for('UserService'),

  UserRepository: Symbol.for('UserRepository'),

  UserEntityConverter: Symbol.for('UserEntityConverter'),
}

export { TYPES }
