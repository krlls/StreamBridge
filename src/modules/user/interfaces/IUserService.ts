import { CreateUserDTO } from '../dtos/CreateUserDTO'
import { UserDTO } from '../dtos/UserDTO'

export interface IUserService {
  createUser: (createUser: CreateUserDTO) => Promise<UserDTO | undefined>,
  findUserById: (userId: number) => Promise<UserDTO | undefined>,
}
