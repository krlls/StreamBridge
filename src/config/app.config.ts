import { ServerConfig } from '../types/TServer'
import { E_NODE_ENV } from '../types/common'
import { NODE_ENV } from './env'

export const serverConfig: ServerConfig = {
  port: 3000,
  silent: NODE_ENV === E_NODE_ENV.TEST,
}
