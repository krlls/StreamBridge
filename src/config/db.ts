import { DataSourceOptions } from 'typeorm/data-source/DataSourceOptions'

import { E_NODE_ENV } from '../types/common'

const entitiesPatch = 'src/infra/db/Sqlite/entities/**/*.ts'
const dbPatch = 'data/db.sqlite3'
export const sqlLite: Record<E_NODE_ENV, DataSourceOptions> = {
  [E_NODE_ENV.TEST]: {
    type: 'better-sqlite3',
    database: ':memory:',
    entities: [entitiesPatch],
    synchronize: true,
  },
  [E_NODE_ENV.DEV]: {
    type: 'better-sqlite3',
    database: dbPatch,
    entities: [entitiesPatch],
    synchronize: true,
    migrationsRun: false,
    logging: true,
    enableWAL: true,
  },
  [E_NODE_ENV.PROD]: {
    type: 'better-sqlite3',
    database: dbPatch,
    entities: [entitiesPatch],
    synchronize: false,
    migrationsRun: true,
    logging: false,
    enableWAL: true,
  },
}
