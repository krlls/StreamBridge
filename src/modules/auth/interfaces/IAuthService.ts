import { LoginDTO } from '../dtos/LoginDTO'
import { ServiceResultDTO } from '../../../types/common'
import { LoginSuccessDTO } from '../dtos/loginSuccessDTO'

export interface IAuthService {
  login: (login: LoginDTO) => Promise<ServiceResultDTO<LoginSuccessDTO>>,
}
