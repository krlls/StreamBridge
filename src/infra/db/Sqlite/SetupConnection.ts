import { DataSource, EntityTarget, Repository } from 'typeorm'
import Database from 'better-sqlite3'

import { NODE_ENV, sqlLite } from '../../../config'

export const dataSource = new DataSource(sqlLite[NODE_ENV])

export class SqliteDB {
  private static _instance: SqliteDB

  public static get instance(): SqliteDB {
    if (!this._instance) this._instance = new SqliteDB()

    return this._instance
  }

  private dbConnect!: DataSource
  private testdb!: any

  async setupTestDB() {
    this.testdb = new Database('')
    this.dbConnect = await dataSource.initialize()
  }

  async teardownTestDB() {
    await this.dbConnect.destroy()
    this.testdb.close()
  }
}

export const getRepository = <Entity extends Record<string, any>>(target: EntityTarget<Entity>): Repository<Entity> => {
  return dataSource.getRepository(target)
}

export default dataSource
