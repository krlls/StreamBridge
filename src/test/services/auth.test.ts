import { expect, test, describe, beforeEach, afterEach } from '@jest/globals'

import { SqliteDB } from '../../infra/db/Sqlite/SetupConnection'
import { fakeControllerContainer } from '../helpers/inversify.test.config'
import { TYPES } from '../../types/const'
import { IAuthService } from '../../modules/auth/interfaces/IAuthService'
import { IUserService } from '../../modules/user/interfaces/IUserService'
import { CreateUserDTO } from '../../modules/user/dtos/CreateUserDTO'
import { LoginDTO } from '../../modules/auth/dtos/LoginDTO'
import { testUserData } from '../helpers/test.helpers'

const authService = fakeControllerContainer.get<IAuthService>(TYPES.AuthService)
const userService = fakeControllerContainer.get<IUserService>(TYPES.UserService)
describe('Auth service tests', () => {
  const user = new CreateUserDTO(testUserData)

  beforeEach(async () => {
    await SqliteDB.instance.setupTestDB()
    await userService.createUser(user)
  })

  afterEach(async () => {
    await SqliteDB.instance.teardownTestDB()
  })

  test('Auth works', async () => {
    const loginData = new LoginDTO(user.login, user.pass)

    const result = await authService.login(loginData)

    expect(result).toHaveProperty('token')
  })

  test('Auth wrong credentials', async () => {
    const loginData = new LoginDTO(user.login, user.pass + 'random value 123')

    const result = await authService.login(loginData)

    expect(result).not.toHaveProperty('token')
    expect(result).toHaveProperty('error')
    expect(result).toMatchSnapshot()
  })
})
