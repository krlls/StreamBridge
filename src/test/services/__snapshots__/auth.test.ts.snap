// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth tests Auth wrong credentials 1`] = `
ErrorDTO {
  "error": "Inkorrect credentials",
}
`;
